runtime/src/main/java/org/corfudb/util/FileWatcher.java

```diff
@@ -122,7 +136,10 @@ public void close() {
         isStopped.set(true);
 
         try {
-            this.watchService.close();
+            if (watchService != null) {
+                watchService.close();
+            }
+            isRegistered.set(false);
```

Comment:
nit: move this to before `watchService.close();` is called

Reply:
CorfuRuntime invokes `close()` in its `stop()` method -
 `sslCertWatcher.ifPresent(FileWatcher::close);`
 
It does not look good to set `isRegistered` to false before `close()` is called in CorfuRuntime class. 
Hence, I felt it is better to set it inside `close()` method compared to before.

In fact, I found this bug while writing unit tests and ITs. :)